protobuf_serialization/reader.nim

```diff
@@ -42,7 +43,54 @@ proc readFieldInto[T: object](
       raise (ref ValueError)(msg: "not enough bytes")
     memoryInput(tmp).readValueInternal(value)
 
-proc readFieldInto[T: not object and (seq[byte] or not seq)](
+when defined(ConformanceTest):
+  proc readFieldInto[K, V](
+    stream: InputStream,
+    value: var Table[K, V],
+    header: FieldHeader,
+    ProtoType: type
+  ) =
+    # I know it's ugly, but I cannot find a clean way to do it
+    # ... And nobody cares about map
+    when K is SomePBInt and V is SomePBInt:
+      type
+        TableObject {.proto3.} = object
+          key {.fieldNumber: 1, pint.}: K
+          value {.fieldNumber: 2, pint.}: V
+    elif K is SomePBInt:
+      type
+        TableObject {.proto3.} = object
+          key {.fieldNumber: 1, pint.}: K
+          value {.fieldNumber: 2.}: V
+    elif V is SomePBInt:
+      type
+        TableObject {.proto3.} = object
+          key {.fieldNumber: 1.}: K
+          value {.fieldNumber: 2, pint.}: V
+    else:
+      type
+        TableObject {.proto3.} = object
+          key {.fieldNumber: 1.}: K
+          value {.fieldNumber: 2.}: V
+    var tmp = default(TableObject)
+    stream.readFieldInto(tmp, header, ProtoType)
+    value[tmp.key] = tmp.value
+
+proc readFieldInto[T: enum](
+  stream: InputStream,
+  value: var T,
+  header: FieldHeader,
+  ProtoType: type
+) =
+  when 0 notin T:
+    {.fatal: $T & " definition must contain a constant that maps to zero".}
+  header.requireKind(WireKind.Varint)
+  let enumValue = stream.readValue(ProtoType)
+  if not checkedEnumAssign(value, enumValue.int32) and
+    not checkedEnumAssign(value, 0):
+    raise (ref ValueError)(msg: "Attempted to decode an invalid enum value")
```

Comment:
That can never happen, right?

Reply:
You're right, I added the `when 0 notin T:` later and it makes the `checkedEnumAssign(value, 0)` obsolete.